### ISSUE TITLE
Download ahead while reading still doesn't work.

### ISSUE THREAD

**Issue by THElegend5** [tachiyomiorg/tachiyomi#9669]
### Steps to reproduce

1. Enable download ahead while reading.
2. Start reading a title with atleast 2 chapters downloaded.

### Expected behavior

Tachiyomi downloads ahead the number of chapters ahead as specified in the settings.

### Actual behavior

Tachiyomi doesn't download the chapters.

### Tachiyomi version

r5882

### Android version

Android 13

### Device

Smasung S21 FE

### Other details

Same issue as #9471 .
#9640 didn't seem to fix it.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have gone through the [FAQ](https://tachiyomi.org/help/faq/) and [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by tisserm** [comment]
From what I've observed, the "download ahead" feature is triggered too late. 
Let's say I'm starting a new series, downloaded the first two chapters and set "auto download while reading" to "next 2 unread chapters".
Then the "download ahead" feature will kick in while reading the third chapter and download the fourth and fifth chapters. It'll then be triggered again while reading the sixth chapter and so on.
The expected behaviour is to always have at least two unread chapters downloaded at any point in time

**Comment by THElegend5** [comment]
For me it doesn't seem to be working even with 5 chapters already downloaded.

**Comment by THElegend5** [comment]
@arkon still does'nt seem to be working.

**Comment by arkon** [comment]
Seems to be working fine for me?

**Comment by THElegend5** [comment]
> Seems to be working fine for me?

For me it seems to be triggering but at random also almost always triggers with one downloaded chapter instead of the required atleast two. And also waits for one downloaded chapter to be left to trigger again so doesn't consistently keep 5 chapters downloaded and such.

**Comment by tisserm** [comment]
In my case, it's mostly fixed now. There is still kind of an offset because the app counts unread and already read chapters in the required downloaded chapters (instead of just unread chapters). So occasionally the only two downloaded chapters are the one I just read and the following one. Don't know if the behaviour was the same before the bug though

**Comment by THElegend5** [comment]
> In my case, it's mostly fixed now. There is still kind of an offset because the app counts unread and already read chapters in the required downloaded chapters (instead of just unread chapters). So occasionally the only two downloaded chapters are the one I just read and the following one. Don't know if the behaviour was the same before the bug though 

It isn't, the expected behaviour is for it to trigger somewhere around halfway of the first download woth another chapter already downloaded present after that. I have also consistently found it triggering with one downloaded chapter so even if it is triggering it doesn't seem to be triggering correctly as I mentioned before.

**Comment by BrutuZ** [comment]
> > Seems to be working fine for me?
> 
> For me it seems to be triggering but at random also almost always triggers with one downloaded chapter instead of the required atleast two. And also waits for one downloaded chapter to be left to trigger again so doesn't consistently keep 5 chapters downloaded and such.

That has been my experience as well.
I have it set to download 5 chapters ahead instead of 2, and I also set the auto delete to keep the last read chapter.
I preload the first couple chapters before starting and whenever I go back to the chapter list after however many chapters read I'd expect to have 6 downloaded chapters "at all times" until catching up with the series, instead it's a roll of the dice whether I'll have the expected next 5 chapters downloaded, only the next one or anything in-between

**Comment by AntsyLich** [comment]
Closing in favor of #10151
